extensions/ql-vscode/src/common/vscode/commands.ts

```diff
@@ -0,0 +1,25 @@
+import { commands } from "vscode";
+import { commandRunner } from "../../commandRunner";
+import { CommandFunction, CommandManager } from "../../packages/commands";
+
+/**
+ * Intializes a command manager for VSCode, wrapping the commandRunner
+ * and vscode.executeCommand.
+ */
+export function initializeVSCodeCommandManager<
+  Commands extends Record<string, CommandFunction>,
+>(): CommandManager<Commands> {
+  return new CommandManager(commandRunner, wrappedExecuteCommand);
+}
+
+async function wrappedExecuteCommand<
```

Comment:
Why are we wrapping this? Is it just to satisfy TS and set the expected types? Might be worth adding a code comment to explain.

Also should it be named `wrapExecuteCommand` ?

Reply:
Yeah the types are tricky to make work without a separate function as the `Promise` part does not behave in the most expected way. Which is also why we need the wrapping and unwrapping of `Promise<Awaited`. I'll add a comment.

> Also should it be named wrapExecuteCommand ?
:+1: